### ISSUE TITLE
'Function cannot be evaluated at initial parameters'

### ISSUE THREAD

**Issue by KennedyMurphy** [robjhyndman/forecast#217]
I've ran into an old bug ( #78) for specific data / model combinations with the forecast package (version 6.2). I'm trying to benchmark multiple different ets models for several different time series. A handful of time series fail for certain ets models.

For instance, given
`x <- ts(c(103,27, 114, 50, 140, 956 , 3315 ,3170 ,1052,  762, 1325, 1770, 1263, 2244, 1203, 1052, 1521, 1680 ,1432 , 840,  885, 1209, 1222,  690, 1608,  969, 1004 ,1001,  788 ,1163 ,1340, 1100,  543,  876, 1209), start = c(2012,9), freq = 12)`

The call `ets(x, model = "AMA", restrict = FALSE)` returns the error:

> Error in etsmodel(y, errortype[i], trendtype[j], seasontype[k], damped[l],  : 
>   function cannot be evaluated at initial parameters

I've tracked the error back to the call to Nelder Mead within the etsmodel function:
`.Call("etsNelderMead", par, env, -Inf, sqrt(.Machine$double.eps), 1, 0.5, 2, trace, maxit, package = "forecast")`
 which was the source of the error in issue #78. 

What is interesting is that if I use a frequency of 12.1, 12.01, 12.001, or 12.0001, the above call to ets does not fail. But not with a frequency of 12.00001.

Furthermore, omitting the last data point in x with a frequency of 12 does not cause any errors.

Have I overlook a solution that has been posted or is this still an outstanding problem? Thanks in advance.

**Comment by robjhyndman** [comment]
Fixed in https://github.com/robjhyndman/forecast/commit/73ce3b94c21d02b9d5bde8616d909747d990d1c5

**Comment by robjhyndman** [comment]
First make sure you are using the most recent version of the forecast package on CRAN (8.4). Then, if you still have problems, please provide a minimal reproducible example. See https://robjhyndman.com/hyndsight/minimal-reproducible-examples/ for details.

**Comment by MehrdadVaredi** [comment]
Hi everyone,
The error says 'Function cannot be evaluated at initial parameters', in my case when I looked at the data at the beginning of the time serie, they were NA. So make sure you drop the NAs from the beginning if for some reason after imputation they show up. In my case I had smoothed the curve again and it had caused NAs again!
I hope this solves the issue.
So if it says 'Function cannot be evaluated at initial parameters', actually the data may have caused it, get a head(your dataset) or head(your time serie) and check it out!
Have a good one!

Mehrdad
